dcpy/lifecycle/ingest/configure.py

```diff
@@ -51,28 +61,58 @@ def read_template(
     return Template(**template_yml)
 
 
-def get_version(template: Template, timestamp: datetime) -> str:
-    """Given parsed dataset template, determine version.
+def get_version(source: Source, timestamp: datetime | None = None) -> str:
+    """
+    Given parsed dataset template, determine version.
     If version's source has no custom logic, returns formatted date
-    from provided datetime"""
-    match template.source:
-        case socrata.Source() as socrata_source:
-            return extract_socrata.get_version(socrata_source)
-        case edm.publishing.GisDataset() as gis_dataset:
-            return publishing.get_latest_gis_dataset_version(gis_dataset.name)
+    from provided datetime
+    """
+    match source:
+        case socrata.Source():
+            return extract_socrata.get_version(source)
+        case GisDataset():
+            return publishing.get_latest_gis_dataset_version(source.name)
         case _:
+            if timestamp is None:
+                raise TypeError(
+                    f"Version cannot be dynamically determined for source of type {source.type}"
+                )
             return timestamp.strftime("%Y%m%d")
 
 
-def get_config(
-    template: Template, version: str, timestamp: datetime, file_name: str
-) -> Config:
-    """Simple wrapper to produce a recipes ExtractConfig from a parsed template
-    and other computed values"""
+def get_filename(source: Source, ds_name: str) -> str:
```

Comment:
is the output of the fn equivalent to `raw_dataset` name attribute in Config object?

Reply:
Yes - or more specifically, this is what sets the raw_dataset name attribute